### PR TITLE
Set MSRV, and up to 1.53 due to issues with net library

### PR DESCRIPTION
**Pull request by DarthHater**
There are issues with the net library prior to 1.53, and it is likely we use it, so this move is mostly a) let's just move up to MSRV of 1.53 to avoid a lengthy look at if we are totally affected.

This pull request makes the following changes:
* Build using `1.53` in CircleCI
* Set `rust-version = 1.53` in `Cargo.toml`

Build failure fixed now that we can ignore a vulnerability (done in #57 )

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ readme = "README.md"
 repository = "https://github.com/sonatype-nexus-community/cargo-pants"
 description = "cargo-pants is a cargo subcommand application that provides a bill of materials and a list of which dependencies have a vulnerability, powered by Sonatype OSSIndex"
 license = "Apache-2.0"
+rust = "1.53"
```
Comment (amy-keibler) [inline Cargo.toml]: `cargo check` gives

> warning: unused manifest key: package.rust

Are we concerned about using a cargo feature that is still in the RFC stage, given that the key is an open question? [RFC context here](https://github.com/rust-lang/rust/issues/65262)

**Review by amy-keibler — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (bhamail) [reply Cargo.toml]: Interesting to know. I'm not concerned. "What, me worry?"

**Review by bhamail — APPROVED** [review]
+1

someday I hope to get the "ignore" feature implemented - I believe that is the only thing causing this build to fail.

**Comment by DarthHater** [comment]
@amy-keibler @bhamail I used the suggestion from: https://users.rust-lang.org/t/call-for-testing-unstable-cargo-support-for-rust-version/59386 so we are at least setup based on what looks like is coming

**Review by amy-keibler — APPROVED** [review]

**Pull request merged**